openraft/src/replication/mod.rs

```diff
@@ -387,6 +392,11 @@ impl<D: AppData, R: AppDataResponse, N: RaftNetwork<D>, S: RaftStorage<D, R>> Re
 
         // Continue to find the matching log id on follower.
         self.max_possible_matched_index = conflict.index - 1;
+
+        // log_size == max_payload_entries means there may be more logs to replicate
+        if log_size == self.config.max_payload_entries.try_into().unwrap() {
+            return Err(ReplicationError::MoreLogs);
+        }
```

Comment:
Nice catch My hero!

IMHO, there is another related issue that when keeping sending logs, the ReplicationStream should also keep draining messages from `self.repl_rx`.
E.g., when a Leader reverts to Follower, the replication should stop at once.

Thus it may be better if we do not use an Error to express the **continue sending** state.
But instead, drain `self.repl_rx` and check manually whether there are more log to send.

At line 660, before entering the blocking wait, 

```
// this method will return an error if the Leader is dropped.
self.try_drain_raft_rx().await?;
if self.has_more_log() {
    continue;
}

tokio::select!{ ... }
```

https://github.com/datafuselabs/openraft/blob/64f3b8b9f997f834a6da62803610f6a9ccc62f1e/openraft/src/replication/mod.rs#L650-L681